execution/trigger/watermark.go

```diff
@@ -124,35 +115,45 @@ func (wt *WatermarkTrigger) PollKeyToFire(ctx context.Context, tx storage.StateT
 	if err == nil {
 		watermark = octoWatermark.AsTime()
 	} else if err != storage.ErrNotFound {
-		return octosql.ZeroValue(), errors.Wrap(err, "couldn't get current watermark")
+		return nil, errors.Wrap(err, "couldn't get current watermark")
 	}
 
-	if watermark.Before(sendTime) {
-		panic("unreachable")
+	keys, times, err := timeKeys.GetUntil(watermark)
+	if err != nil {
+		if err == storage.ErrNotFound {
+			panic("unreachable")
+		}
+		return nil, errors.Wrap(err, "couldn't get keys by time until watermark")
 	}
 
-	err = timeKeys.Delete(key, sendTime)
-	if err != nil {
-		return octosql.ZeroValue(), errors.Wrap(err, "couldn't delete key")
+	for i := range keys {
+		if watermark.Before(times[i]) {
```

Comment:
If it's truly unreachable (as when one adds this as a default in a switch) then why add this if statement?
Is this like a sanity check?

Reply:
Yeah, it's a sanity check.

Reply:
Cause I think this would be hell to debug if it didn't work.